### DISCUSSION TITLE
The pyodide web editor URL sharing is awesome! It made me want more features!

### DISCUSSION THREAD

**Discussion by villares** [berinhard/pyp5js#210] [Ideas]
So, now that I'm using it on some of my classes, I wish I had a "present mode", like full screen canvas only mode we could use to show off sketches (and maybe embed them with iframes somewhere else).

It could be triggered by a button or by some argument in the URL and maybe both.

cc @jvfe 

Cheers,
Alexandre

**Comment by jvfe** [comment]
Hi, do you mean something like what the p5.js editor does (i.e. https://editor.p5js.org/p5/full/VwIq-vp0LDL)? I think it would be very doable to have a subpage (like https://berinhard.github.io/pyp5js/pyodide/full/) where it would only display the canvas. 

It would be quite a bit more work than the url sharing feature but still doable - I'll try to get to it when I have the time.

**Reply by villares** [reply]
Yeah! That would be awesome.

Would an extra argument in the URL make it easier?
Something like: 
`https://berinhard.github.io/pyp5js/pyodide/?full=true&sketch=ZGVmJTIwc...`
https://berinhard.github.io/pyp5js/pyodide/?full=true&sketch=ZGVmJTIwc2V0dXAoKSUzQSUwQSUyMCUyMCUyMCUyMGNyZWF0ZUNhbnZhcygyMDAlMkMlMjAyMDApJTBBJTBBZGVmJTIwZHJhdygpJTNBJTBBJTIwJTIwJTIwJTIwYmFja2dyb3VuZCgyMDApJTBBJTIwJTIwJTIwJTIwZGlhbWV0ZXIlMjAlM0QlMjBzaW4oZnJhbWVDb3VudCUyMCUyRiUyMDYwKSUyMColMjA1MCUyMCUyQiUyMDUwJTBBJTIwJTIwJTIwJTIwZmlsbCglMjJibHVlJTIyKSUwQSUyMCUyMCUyMCUyMGVsbGlwc2UoMTAwJTJDJTIwMTAwJTJDJTIwZGlhbWV0ZXIlMkMlMjBkaWFtZXRlciklMEElMjAlMjAlMjAlMjA%3D

Cheers, João Vitor!

**Reply by villares** [reply]
but I can also imagine the subpage (maybe under /full/) that has no visible editor, just picks the code from the url argument and runs it

**Reply by villares** [reply]
So, here is my proof of concept! :smile: 

https://abav.lugaralgum.com/testing/?sketch=ZGVmJTIwc2V0dXAoKSUzQSUwQSUyMCUyMCUyMCUyMGNyZWF0ZUNhbnZhcygxMDAwJTJDJTIwOTAwKSUwQSUwQWRlZiUyMGRyYXcoKSUzQSUwQSUyMCUyMCUyMCUyMGJhY2tncm91bmQoMjAwKSUwQSUyMCUyMCUyMCUyMGRpYW1ldGVyJTIwJTNEJTIwc2luKGZyYW1lQ291bnQlMjAlMkYlMjA2MCklMjAqJTIwaGVpZ2h0JTIwJTJGJTIwMiUwQSUyMCUyMCUyMCUyMGZpbGwoJTIyZ3JlZW4lMjIpJTBBJTIwJTIwJTIwJTIwZWxsaXBzZSh3aWR0aCUyMCUyRiUyMDIlMkMlMjBoZWlnaHQlMjAlMkYlMjAyJTJDJTIwZGlhbWV0ZXIlMkMlMjBkaWFtZXRlciklMEElMjAlMjAlMjAlMjA%3D#executeBtn

I just deleted stuff and hid the code editor!

I'm sure you clever JavaScript-knowing people can remove the run button and make it all run in the event of finishing loanding pyodide... I guess.

**Comment by MrMenezes** [comment]
Like this?
https://mrmenezes.github.io/spiral/

**Reply by villares** [reply]
This is nice also, but not quite what I mean. You did not keep João Vitor's machinery for decodng code from the URL.

The idea is to provide, from the editor, a way of making a sharable link to a page (like yours, or like mine) with the running sketch.
- You got the full-screen and running immediately part.
- My example gets the customizable, shareable-in-the-url-code part
  (but I couldn't remove the run button, and I had to manually manipulate my URL from the default sharing link button).

Does it make any sense, @MrMenezes?

Cheers!
